crates/parcel_plugin_transformer_js/src/transformer.rs

```diff
@@ -69,7 +87,22 @@ impl TransformerPlugin for ParcelJsTransformerPlugin {
       return Err(anyhow!(format!("{:#?}", errors)));
     }
 
-    let asset = Asset::new_empty(input.file_path().to_path_buf(), source_code);
+    let file_path = input.file_path();
+    let asset_type = FileType::from_extension(
+      file_path
+        .extension()
+        .and_then(|s| s.to_str())
+        .unwrap_or_default(),
+    );
+
+    let asset = Asset {
+      asset_type,
+      code: source_code,
+      env: env.clone(),
+      file_path: file_path.to_path_buf(),
+      ..Asset::default()
```

Comment:
I'm not 100% sure, but if there's any properties that can be set by previous transformers this will override them with the defaults.